tests/utils/test_kdata.py

```diff
@@ -0,0 +1,229 @@
+"""Tests of utilities for kdata objects."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+from einops import rearrange
+from einops import repeat
+
+from mrpro.data import AcqInfo
+from mrpro.data import KData
+from mrpro.data import KHeader
+from mrpro.data import KTrajectory
+from mrpro.utils import modify_acq_info
+from mrpro.utils.kdata import combine_k2_k1_into_k1
+from mrpro.utils.kdata import sel_other_subset_from_kdata
+from mrpro.utils.kdata import split_idx
+from mrpro.utils.kdata import split_k1_into_other
+from mrpro.utils.kdata import split_k2_into_other
+from tests.conftest import RandomGenerator
+from tests.conftest import generate_random_data
+from tests.conftest import generate_random_trajectory
+from tests.conftest import random_acquisition
+from tests.conftest import random_full_ismrmrd_header
+
+
+@pytest.fixture(params=({'seed': 0, 'nother': 10, 'nk2': 40, 'nk1': 20},))
+def random_kheader_shape(request, random_acquisition, random_full_ismrmrd_header):
+    """Random (not necessarily valid) KHeader with defined shape."""
```

Comment:
what happens when KHeader is not valid? Isn't the test failing then due to an exception being thrown?

Reply:
random -> not all values match, most combinations of values are physically impossible.
but for most test purposes, most fields can indeed be random. this is what we use here.